jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -244,20 +246,31 @@ private int getActionSize()
     @Override
     public void dump(Appendable out, String indent) throws IOException
     {
-        super.dump(out, indent);
         Selector selector = _selector;
+        List<String> keys = null;
+        List<Runnable> actions = null;
         if (selector != null && selector.isOpen())
         {
-            List<Runnable> actions;
+            DumpKeys dump = new DumpKeys();
+            String actionsAt;
             try (Locker.Lock lock = _locker.lock())
             {
+                actionsAt = StdErrLog.timestamp(System.currentTimeMillis(),TimeUnit.MILLISECONDS);
                 actions = new ArrayList<>(_actions);
+                _actions.addFirst(dump);
+                _selecting = false;
             }
-            List<Object> keys = new ArrayList<>(selector.keys().size());
-            DumpKeys dumpKeys = new DumpKeys(keys);
-            submit(dumpKeys);
-            dumpKeys.await(5, TimeUnit.SECONDS);
-            dump(out, indent, Arrays.asList(new DumpableCollection("keys", keys), new DumpableCollection("actions", actions)));
+            _selector.wakeup();
+            keys = dump.get(5, TimeUnit.SECONDS);
+            String keysAt = StdErrLog.timestamp(System.currentTimeMillis(),TimeUnit.MILLISECONDS);
+            if (keys==null)
+                keys = Collections.singletonList("NO DUMP RESPONSE");
```

Comment:
Improve the message... Must have normal case and explain that it's the keys dump that we did not get a response from, and add how long we waited, etc.